cachito/workers/pkg_manager.py

```diff
@@ -0,0 +1,58 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+import logging
+import os
+import subprocess
+import tarfile
+import tempfile
+
+from cachito.errors import CachitoError
+from cachito.workers.config import get_worker_config
+
+
+log = logging.getLogger(__name__)
+
+
+def resolve_gomod_deps(archive_path):
+
+    with tempfile.TemporaryDirectory(prefix='cachito-') as temp_dir:
+        source_dir = _mount_app_src(archive_path, temp_dir)
+        env = {
+            'GOPATH': temp_dir,
+            'GO111MODULE': 'on',
+            'GOCACHE': temp_dir,
+            'GOPROXY': get_worker_config().athens_url,
+            'PATH': os.environ.get('PATH', ''),
+        }
+        cmd = ('go', 'list', '-m', 'all')
+
+        go_list = subprocess.Popen(cmd, stdout=subprocess.PIPE, stderr=subprocess.PIPE,
+                                   text=True, encoding='utf-8', env=env, cwd=source_dir)
```

Comment:
Bummer, the `test` parameter was only introduced in python 3.7. It seems that the existing `universal_newlines` will do the same. I'll change it.

https://docs.python.org/3/library/subprocess.html#subprocess.Popen

Reply:
Fixed!